dohq_artifactory/admin.py

```diff
@@ -311,6 +335,92 @@ def groups(self, value):
     def groups(self):
         self._groups = []
 
+    @property
+    def api_key(self):
+        return self._ApiKeyHandler(self)
+
+    class _ApiKeyHandler:
+        def __init__(self, user):
+            """
+            :param user: User instance
+            """
+            self._user = user
+            self.url = "/api/security/apiKey"
+
+        def __repr__(self):
+            return self.get() or ""
+
+        def __str__(self):
+            return self.get() or ""
+
+        def get(self):
+            """
+            Get an API key for the current user
+            :return: (str) API key
+            """
+            response = self._user._authenticated_user_request(
+                api_url=self.url,
+                request_type=self._user._session.get,
+            )
+            api_key = json.loads(response).get("apiKey", None)
+
+            return api_key
+
+        def create(self):
+            """
+            Create an API key for the current user.
+            Returns an error if API key already exists - use api_key_regenerate to regenerate API key instead.
+            :return: (str) API key
+            """
+            response = self._user._authenticated_user_request(
+                api_url=self.url,
+                request_type=self._user._session.post,
+            )
+
+            api_key = json.loads(response)["apiKey"]
+
+            return api_key
+
+        def regenerate(self):
+            """
+            Regenerate an API key for the current user
+            :return: (str) API key
+            """
+            if self._user.api_key is None:
+                raise ArtifactoryException(
+                    "API key does not exist for {}. Please use api_key.create".format(
+                        self._user.name
+                    )
+                )
+
+            response = self._user._authenticated_user_request(
+                api_url=self.url,
+                request_type=self._user._session.put,
+            )
+            api_key = json.loads(response).get("apiKey", None)
```

Comment:
`.get("apiKey", None)` when is the case possible? 
May be we should raise an error in this case - when you try to regenerate and gets no key back

Reply:
I changed return signature for each function. We always return string but in case of no key or some issue in parsing response it will be empty string. Then user can handle it by themselves. I think from user perspective it is easier to check `if api_key` rather than do `except ArtifactoryException` and parse error message to ensure that we failed in getting the key